### PR TITLE
[WIP] Adapt Account to new schema Account

### PR DESCRIPTION
**Pull request by iambmelt**
Needed for sharing SSO state

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java [hunk 1]
```diff
@@ -153,6 +159,74 @@ public RefreshToken createRefreshToken(
...
+        // Required fields
+        refreshTokenOut.setUniqueUserId("");
+        refreshTokenOut.setEnvironment("");
+        refreshTokenOut.setCredentialType(CredentialType.RefreshToken.name());
+        refreshTokenOut.setClientId("");
+        refreshTokenOut.setSecret(refreshTokenIn.getRefreshToken());
+
+        // Optional fields
+        refreshTokenOut.setTarget("");
+        refreshTokenOut.setCachedAt("");
```
Comment (shoatman) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java]: Caching logic can set this.

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java [hunk 2]
```diff
@@ -153,6 +159,74 @@ public RefreshToken createRefreshToken(
...
+        refreshTokenOut.setExpiresOn("");
+        refreshTokenOut.setClientInfo("");
+        refreshTokenOut.setFamilyId("");
+        refreshTokenOut.setUsername("");
```
Comment (shoatman) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java]: Is this necessary.  Doesn't seem like it

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java [hunk 3]
```diff
@@ -153,6 +159,74 @@ public RefreshToken createRefreshToken(
...
+
+        // TODO populate this object
+
+        // Required fields
+        refreshTokenOut.setUniqueUserId("");
+        refreshTokenOut.setEnvironment("");
+        refreshTokenOut.setCredentialType(CredentialType.RefreshToken.name());
+        refreshTokenOut.setClientId("");
+        refreshTokenOut.setSecret(refreshTokenIn.getRefreshToken());
+
```
Comment (shoatman) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java]: I"d all required fields to the fresh token object.  Base class.... with the exception of credential type....

File: common/src/main/java/com/microsoft/identity/common/internal/cache/AccountCredentialCacheKeyValueDelegate.java [hunk 4]
```diff
@@ -73,21 +73,24 @@ public String generateCacheKey(Account account) {
         return cacheKey;
     }
 
-    private String generateCacheValueInternal(final AccountCredentialBase baseObject) {
+    private String generateCacheValueInternal(final Object baseObject) {
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/AccountCredentialCacheKeyValueDelegate.java]: Not super thrilled about this change, but not seeing a great way around it yet

**Comment by iambmelt** [comment]
Need to reconcile the behavior of #71 since it's forcing API changes to constructors/interfaces

**Comment by iambmelt** [comment]
Notes for Monday sync w/ @addev-ashish @shoatman 
- Behavior of #71 impacts API of `IAccountCredentialAdapter`, constructors for `IDToken` (& subclasses) -- has trickle-down effects on the following:
    * `MicrosoftStsRefreshToken`
    * `MicrosoftStsOauth2Strategy`
    * `AzureActiveDirectoryRefreshToken`
    * `AzureActiveDirectoryOauth2Strategy`
    * `MicrosoftStsAccountCredentialAdapter`

Basically blocked until this is resolved...

In short:
IMO at some level we need to be able to rely on these fields -- they're constructed enough times that catching these errors and doing something sensible with them is increasingly difficult

**Comment by iambmelt** [comment]
Going to merge this to simplify bringing test fixes up from `dev`
